cmd/charm/charmcmd/push.go

```diff
@@ -141,16 +141,18 @@ func (c *pushCommand) Run(ctxt *cmd.Context) error {
 	case c.id.Series != "":
 		ch, err = charm.ReadCharmDir(srcDir)
 	default:
-		ch, err = charm.ReadCharmDir(srcDir)
-		if err == nil {
-			break
-		}
-		if b1, err1 := charm.ReadBundleDir(srcDir); err1 == nil {
-			b = b1
-			err = nil
-			c.id.Series = "bundle"
+		if charm.IsCharmDir(srcDir) {
+			ch, err = charm.ReadCharmDir(srcDir)
+			if err == nil {
```

Comment:
Did you intend to do:

     if charm.IsCharmDir(srcDir) {
        ch, err = charm.ReadCharmDir(srvDir)
        break
    }
    if b, err = charm.ReadBundleDir(srvDir); err == nil {
        c.id.Series = "bundle"
    }


?

Although I actually don't see why we're setting c.id.Series here at all - why not just do:

     if charm.IsCharmDir(srcDir) {
        ch, err = charm.ReadCharmDir(srvDir)
     } else {
         b, err = charm.ReadBundleDir(srvDir)
     }

and set c.id.Series inside the b != nil case below?